File: src/userSummary.ts
import { Component, Input, Output, EventEmitter } from "@angular/core";
@Component({
    selector: "user-summary",
    templateUrl: "src/userSummary.html"
})
export class UserSummary {
    // Input + "Change"=Output
    @Input() firstName: string;
    @Output() firstNameChange: EventEmitter<string> = new EventEmitter<string>();
    public onChanged() {
        this.firstNameChange.emit(this.firstName);
    }
}

File: App.Service/UserSummary.cs
namespace App.Service
{
    public class UserSummary
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public UserSummary(string firstName, string lastName)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
        }
    }
}